torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1550,10 +1550,64 @@ def _verify(expr, potential_expr):
 
         # 2. Every guard must evaluate to True (but remember many guards
         #    like s0 == s1*2 because trivial due to simplification)
-        for g, tb in self.guards:
-            if self._maybe_evaluate_static(g) is not None:
+        def simplify_until_no_change(expr: sympy.Expr) -> sympy.Expr:
+            new_expr = self.simplify(expr)
+            while new_expr != expr:
```

Comment:
we should also have a max number of iterations that we are willing to do here. Also, we need to audit the preexisting rules to make sure they're terminating, and we need to update the invariants for simplify to say that they need to be terminating.